packages/omgidl-serialization/src/DeserializationInfoCache.ts

```diff
@@ -65,13 +75,23 @@ export type PrimitiveTypeDeserInfo =
   | PrimitiveDeserializationInfo
   | PrimitiveArrayDeserializationInfo;
 
-export type FieldDeserializationInfo = {
+export type FieldDeserializationInfo<
+  DeserInfo extends ComplexDeserializationInfo | PrimitiveTypeDeserInfo =
+    | ComplexDeserializationInfo
+    | PrimitiveTypeDeserInfo,
+> = {
   name: string;
   type: string;
-  typeDeserInfo: ComplexDeserializationInfo | PrimitiveTypeDeserInfo;
+  typeDeserInfo: DeserInfo;
   isArray?: boolean;
   arrayLengths?: number[];
   definitionId?: number;
+  /** Optional fields show undefined if not present in the message.
+   * Non-optional fields show a default value per the spec:
+   * https://www.omg.org/spec/DDS-XTypes/1.3/PDF 7.2.2.4.4.4.7
+   */
+  isOptional: boolean;
+  isComplex: DeserInfo extends ComplexDeserializationInfo ? true : false;
```

Comment:
Are we not missing a defaultvalue?: unknown here? I think we can store the default value here as well?